in_toto/runlib.py

```diff
@@ -708,13 +723,12 @@ def in_toto_record_start(step_name, material_list, signing_key=None,
       lstrip_paths=lstrip_paths)
 
   LOG.info("Creating preliminary link metadata...")
-  environment = {}
   if record_environment:
```

Comment:
I think _one_ of `environment` and `record_environment` should be used. Any practical danger `record_environment` overwrites the `workdir` passed in explicitly using `environment`?

Reply:
cc @lukpueh

Reply:
IMO a passed `environment` should override the recorded environment. This should allow for the same use cases, and we wouldn't need any mutual exclusion handling.